r/R/dataset-write.R

```diff
@@ -53,6 +53,17 @@
 #' - `codec`: A [Codec] which will be used to compress body buffers of written
 #'   files. Default (NULL) will not compress body buffers.
 #' @return The input `dataset`, invisibly
+#' @examples
+#' \donttest{
+#' # we can group by cyl, cyl and gear or even more variables
+#' write_dataset(mtcars, tempdir(), "feather", partitioning = "cyl"))
```

Comment:
sure, I' m using "parquet", because if I leave it blank I see
```
Error in if (format %in% c("csv", "text") || any(opt_names %in% c("delim",  : 
  missing value where TRUE/FALSE needed
```

Reply:
https://issues.apache.org/jira/browse/ARROW-11582 is relevant here (we should make that error message nicer, and possibly default to parquet if we can without hurting anything else / needing to do a scan of the data / etc)